### ISSUE TITLE
ES6 arrow functions

### ISSUE THREAD

**Issue by sebakerckhof** [johnpapa/angular-styleguide#573]
Style Y032 ( https://github.com/johnpapa/angular-styleguide#style-y032 ) recommends using a capture variable for `this`. The main motivation is to avoid confusion with of the context of `this`.

But what is recommended if you look at ES6. Would it be recommended to use arrow functions to avoid confusion about `this`, or still use a capture variable?

When using arrow functions it does get hard to follow https://github.com/johnpapa/angular-styleguide#style-y033 since arrow functions only work as function expressions...

**Comment by johnpapa** [comment]
you can use => in ES6 and TypeScript for `this` captures.

its not in here as this style guide works regardless if you use es5 or 6 or 7

**Comment by rhinoceraptor** [comment]
This should be re-opened, as of 1.5.0 if you use arrow functions as your controller, the controller invocation will throw an error. [Here](http://plnkr.co/edit/GEkdY4Dw11gPIbxIyl09) is a plunkr demonstrating the problem (click 'Run').

Angular makes heavy use of prototypes in its dependency injection. Arrow functions can't be ```new```ed, and they have no ```prototype```. I think a warning not to use them as controller, filter, factory, etc. functions should be added to the ```Named vs anonymous functions``` section ([Y024](https://github.com/johnpapa/angular-styleguide/tree/master/a1#style-y024))

**Comment by gkalpak** [comment]
Angular does not make heavy use of prototypes in it's dependency. Controllers and services (registered using `.service()`) are by definition constructor function. This means the need to be `new`d.
This doesn't have anything to do with Angular's DI system or prototypes.

Also, by definition, arrow functions are not `new`able, thus can't be used as constructors (agant this is not Angular-specific).
Using them as factories or filters (or an other non-`new`able type) should be fine though.
